server/src/main/java/org/elasticsearch/common/settings/SecureSettingsLoader.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.common.settings;
+
+import org.elasticsearch.cli.Terminal;
+import org.elasticsearch.env.Environment;
+
+import java.util.Optional;
+
+/**
+ * An interface for implementing {@link SecureSettings} loaders, that is, implementations that create, initialize and load
+ * secrets stores.
+ */
+public interface SecureSettingsLoader {
```

Comment:
Can this be moved to server-cli? It isn't needed by the main ES process right?

Reply:
Yes you are right, I can move it to the CLI package.

Reply:
I mean I can move it to common.cli, where KeystoreAwareCommand is, server-cli is not possible since we share this with the WindowsServiceDaemon.